core/client/fs/src/test/java/alluxio/client/file/BaseFileSystemTest.java

```diff
@@ -452,37 +490,114 @@ public void unmountException() throws Exception {
   }
 
   /**
-   * Ensures Warnings are logged or exceptions are thrown when an {@link AlluxioURI} with a
-   * scheme://host is passed.
+   * Ensures warnings are logged and exception is thrown when an {@link AlluxioURI} with an invalid
+   * authority is passed.
    */
   @Test
-  public void UriCheckTest() throws Exception {
+  public void uriCheckBadAuthority() throws Exception {
     Configuration.set(PropertyKey.MASTER_HOSTNAME, "localhost");
     Configuration.set(PropertyKey.MASTER_RPC_PORT, "19998");
+    TestAppender appender = new TestAppender();
+    Logger.getRootLogger().addAppender(appender);
+
     AlluxioURI uri = new AlluxioURI("alluxio://localhost:1234/root");
-    boolean exceptionRaised = true;
     try {
-      BaseFileSystem.UriCheck(uri);
-      exceptionRaised = false;
+      mFileSystem.createDirectory(uri);
       fail("Should have failed on bad host and port");
     } catch (IllegalArgumentException e) {
-      assertTrue(exceptionRaised);
+      assertTrue(e.getMessage().contains(
+          "The URI host and port do not match the configured value"));
     }
 
-    uri = new AlluxioURI("alluxio://localhost:19998/root");
-    try {
-      BaseFileSystem.UriCheck(uri);
-    } catch (IllegalArgumentException e) {
-      fail("Should not have thrown an exception with a valid scheme://host:port combination.");
+    for (LoggingEvent e : appender.mEvents) {
+      if (e.getLevel().equals("WARN")) {
+        assertTrue(e.getRenderedMessage().contains("The URI schemes")
+            || e.getRenderedMessage().contains("The URI authority"));
+      }
     }
 
-    uri = new AlluxioURI("/root");
+    Logger.getRootLogger().removeAppender(appender);
+    ConfigurationTestUtils.resetConfiguration();
```

Comment:
can drop these into `@After` so that they are still run even if an exception is thrown. Even if a test doesn't modify configuration, it doesn't hurt to reset afterwards.